pkg/image/containerdregistry/builder/actions.go

```diff
@@ -0,0 +1,111 @@
+package builder
+
+import (
+	"context"
+	"fmt"
+	"github.com/containerd/containerd/platforms"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"strings"
+)
+
+func (i *imageBuilder) SetLabels(ctx context.Context, platformMatcher platforms.MatchComparer, labels map[string]string) error {
+	ctx = ensureNamespace(ctx)
+
+	var action string
+	if len(labels) > 0 {
+		action := "LABEL"
+		for k, v := range labels {
+			action = fmt.Sprintf("%s %s=%s", action, k, v)
+		}
+	}
```

Comment:
Nit: a `strings.Builder` would be slightly better here because it minimizes memory copying and allocations. This is likely never a memory intensive operation, so no big deal.